### PR TITLE
SAML Certificate Rollover

### PR DESCRIPTION
**Pull request by omgitsbillryan**
Turns our `/saml/metadata` endpoint from having 1 certificate, to 2.  Example:

```xml
<?xml version="1.0"?>
<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" ID="_bbd79f27-f6d8-4379-b017-fe7200d31737" entityID="saml-rp.vetsgov.localhost">
  <md:SPSSODescriptor AuthnRequestsSigned="true" WantAssertionsSigned="false" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
    <md:KeyDescriptor use="signing">
      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
        <ds:X509Data>
          <ds:X509Certificate>
MIIDkDCCAngCCQDJs5zziPV+JDANBgkqhkiG9w0BAQsFADCBiTELMAkGA1UEBhMCVVMxCzAJBgNVBAgTAkRDMRMwEQYDVQQHEwpXYXNoaW5ndG9uMRswGQYDVQQKExJWQSBEaWdpdGFsIFNlcnZpY2UxGjAYBgNVBAMTEXZldHNnb3YubG9jYWxob3N0MR8wHQYJKoZIhvcNAQkBFhBkc3ZhQGV4YW1wbGUuZ292MB4XDTE2MDgwMzE4MTU0MVoXDTE3MDgwMzE4MTU0MVowgYkxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJEQzETMBEGA1UEBxMKV2FzaGluZ3RvbjEbMBkGA1UEChMSVkEgRGlnaXRhbCBTZXJ2aWNlMRowGAYDVQQDExF2ZXRzZ292LmxvY2FsaG9zdDEfMB0GCSqGSIb3DQEJARYQZHN2YUBleGFtcGxlLmdvdjCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALdF4WSDFxSyTL36DXANlT3Akgm4lsnKJauBdb1l4Nl0Xl7vFIfFPfqYVEUxN0B57BG/GWUmpMHJFR5aXz10/DEcYcP1vlLaO2AWKoqAIyv3eF7yHo52lyYGv+jbVsI0CTl03gYIBvsh7e7778RWFOeJzQaNKr11sYm4Rvx48qyXrkd74YqRuRxFE1JlNicKy4cnDgYb7S77RzLpmD95Z07QYceV7KHwNdWgXZd3zGjfI7O8bkTvSOUcH7Bm23bwX7qrlNMYy58B4VI/YfbBSbTo9wYqCln+gdpDiOymX6db4yC4DBjUwD8tcRO249KKinSbgkwSqv+3eoq1XiTSlC8CAwEAATANBgkqhkiG9w0BAQsFAAOCAQEAmISuOhQD8SdkrkPVBWU5wwVGujF+Chm2A2rbBjESFb/f2yoNhYVNdKQhoRY9RPEJRYNypTyFLB9z+wtrQaRviHVKt/b1SfGeSFUMIkYV5/g/WFlQjD0vriyi8Um2V2BuVF5EtWat/mRYLOrHnIePbmokVh+bOS2cYTgXMB1wpAEdV0HQjUlwdNTdyPhBOw3ZiGa7DV3EbVRTq6CqD1IWRzkt3pR9g/T6nLRIlvtFaGwAM9iEi72FJK7zNqarWioCd9n7/NtspL+HdRnm3wd2/s46EbpP5me31EhrQUvEGNxw12ogXTrZ90icrfzPO3GEtc3XxYzO2Zp9LEK6IfZCAA==
</ds:X509Certificate>
        </ds:X509Data>
      </ds:KeyInfo>
    </md:KeyDescriptor>
    <md:KeyDescriptor use="signing">
      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
        <ds:X509Data>
          <ds:X509Certificate>
MIICUDCCAbmgAwIBAgIBADANBgkqhkiG9w0BAQ0FADBFMQswCQYDVQQGEwJ1czELMAkGA1UECAwCTUExFDASBgNVBAoMC0JpbGxzIEhvdXNlMRMwEQYDVQQDDApiaWxsc2hvdXNlMB4XDTE3MTAwNTE1MTU1MVoXDTE4MTAwNTE1MTU1MVowRTELMAkGA1UEBhMCdXMxCzAJBgNVBAgMAk1BMRQwEgYDVQQKDAtCaWxscyBIb3VzZTETMBEGA1UEAwwKYmlsbHNob3VzZTCBnzANBgkqhkiG9w0BAQEFAAOBjQAwgYkCgYEA2soTM3s4bw8UWIQKuDSmmMW4wiOZwqr7xCeHgJ0DfDfOLedUy2xJ4wpeKwf222sUet7tO24gTBVvIlRRx4bEdUfNs+VgZiWb0BcR0IINvVfkGExqDDuQge36Mg3p4KdxHaioeN1rfPWfxeTskDTXrOqIaxMe+omOg0oeO5teIS0CAwEAAaNQME4wHQYDVR0OBBYEFL8OoKeIwcBB9y1NFzDt8Onuj7OoMB8GA1UdIwQYMBaAFL8OoKeIwcBB9y1NFzDt8Onuj7OoMAwGA1UdEwQFMAMBAf8wDQYJKoZIhvcNAQENBQADgYEA0of0/s9XZsQEe9O5ggHUbh7WPd5KWCbWHlOgbqm7QwuENDst7b0RpHSsFPFG5hkxf7Y6/+i7mA3z5QKdC+tnehR51lis6OCOR19iyWMFftqyk/wGUdgxlMeAzN/JQMtiRtECMgdz+NBQ5FxTQEKB0g9hYpP3OcEOn+x7svv5MmQ=
</ds:X509Certificate>
        </ds:X509Data>
      </ds:KeyInfo>
    </md:KeyDescriptor>
    <md:NameIDFormat>
urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified
</md:NameIDFormat>
    <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="http://localhost:3000/auth/saml/callback" index="0" isDefault="true"/>
  </md:SPSSODescriptor>
</md:EntityDescriptor>
```

### CODE REVIEW

File: config/initializers/saml.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # frozen_string_literal: true
 Settings.saml.certificate = File.read(File.expand_path(Settings.saml.cert_path))
 Settings.saml.key = File.read(File.expand_path(Settings.saml.key_path))
+Settings.saml.certificate_new = File.read(File.expand_path(Settings.saml.cert_new_path))
```
Comment (patrickvinograd) [inline config/initializers/saml.rb]: Can we make this tolerant of a nil value in settings, so we don't have to specify a second cert in the steady state where we only have one.

**Comment by omgitsbillryan** [comment]
As a side note, our `vetsgov-internal.cert` expired this past august.  I went ahead and renewed it so that it is still cryptographically connected to `vetsgov-internal.key`.  I'm not entirely sure where we keep the sandbox key & cert these days - I'm still looking for where to update them...

**Comment by omgitsbillryan** [comment]
I need to coordinate more with Tanel to make sure this won't have any adverse results with ID.me.  also, I'd like to test this in staging, which again, requires some coordination with ID.me.

**Comment by patrickvinograd** [comment]
Agreed, let's coordinate testing of this in staging next week.

**Pull request merged**